src/charm.py

```diff
@@ -170,7 +180,10 @@ def _on_zookeeper_pebble_ready(self, event: EventBase) -> None:
         Handles case where workload has shut down due to failing `ruok` 4lw command and
         needs to be restarted.
         """
-        # FIXME: Will need updating when adding in-place upgrade support
+        # don't want to run default pebble ready during upgrades
+        if not self.upgrade_events.idle:
+            return
```

Comment:
Should this be also deferred?

Reply:
If we're upgrading, we want to skip **this** `pebble_ready` call, and use the one in `events.upgrade`